src/txmempool.cpp

```diff
@@ -1036,6 +1039,73 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
     return memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 15 * sizeof(void*)) * mapTx.size() + memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) + memusage::DynamicUsage(vTxHashes) + cachedInnerUsage;
 }
 
+bool CTxMemPool::DumpMempool(FopenFn mockable_fopen_function, bool skip_file_commit) const
+{
+    using SteadyClock = std::chrono::steady_clock;
+
+    auto start = SteadyClock::now();
+
+    std::map<uint256, CAmount> mapDeltas;
+    std::vector<TxMempoolInfo> vinfo;
+    std::set<uint256> unbroadcast_txids;
+
+    static Mutex dump_mutex;
```

Comment:
As a non-`static` or as a `static`? I think as a `static` it may prevent cases where 2 mempools are created with the same persistence path and try to dump at the same time. However, I think that's perhaps somewhat nonsensical anyway. I don't care either way.

Reply:
In that case I'd expect a lock at the filesystem level as opposed to a mutex. But sure, it's not at all a priority, I was just curious because I assumed the reason it was static before is because it wasn't well encapsulated.

Reply:
Followup entry added in umbrella issue #24303

Reply:
This is used consistently in the codebase, see also `DumpBanlist`.